containers/hawkular-metrics-openshift-integration/src/main/java/org/hawkular/openshift/auth/BasicAuthentication.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2014-2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.openshift.auth;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.codec.digest.Md5Crypt;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author mwringe
+ */
+public class BasicAuthentication {
+
+    public static final String BASIC_PREFIX = "Basic ";
+
+    public static final String HTPASSWD_FILE = System.getProperty("hawkular-metrics.openshift.htpasswd-file", "~/" +
+            ".htpasswd");
+
+    public static final String MD5_PREFIX = "$apr1$";
+    public static final String SHA_PREFIX = "{SHA}";
+
+    private static final Logger logger = LoggerFactory.getLogger(BasicAuthentication.class);
+
+    private Map<String, String> users = new HashMap<>();
+
+    public BasicAuthentication() throws Exception {
```

Comment:
Lets not throw too generic exceptions (even just IOException is better), catching Exception catches RuntimeExceptions also.